jdi-light/src/main/java/com/epam/jdi/light/settings/WebSettings.java

```diff
@@ -115,64 +134,220 @@ private static void fillAction(JAction1<String> action, String name) {
+            throw exception("Unknown JDISettings.ELEMENT.smartName: '%s'. Please correct value 'smart.locators.toName' in test.properties." +
+                    "Available names: [%s]", name, print(SMART_MAP_NAME_TO_LOCATOR.keys()));
+        }
+        return Pair.$(name, SMART_MAP_NAME_TO_LOCATOR.get(name));
+    }
+
+    private UseSmartSearch getSmartSearchUse(String prop) {
+        String propLower = prop.toLowerCase().trim().replaceAll(" ", "");
+        switch (propLower) {
+            case "false":
+                return FALSE;
+            case "onlyui":
+                return ONLY_UI;
+            case "uiandelements":
+                return UI_AND_ELEMENTS;
+            case "always":
+                return ALWAYS;
+            default:
+                return UI_AND_ELEMENTS;
+        }
+    }
+
+    private boolean getBoolean(String param) {
         String lowerParams = param.toLowerCase();
         return !lowerParams.equals("off") && !lowerParams.equals("false");
     }
 
-    public static void loadCapabilities(String property, String defaultPath, JAction1<Properties> setCapabilities) {
+    public void loadCapabilities(String property, String defaultPath, JAction1<Properties> setCapabilities) {
         String path = "";
         try {
             path = System.getProperty(property, getProperty(property));
-        } catch (Exception ignore) { }
+        } catch (Exception ignore) {
+        }
         if (isEmpty(path))
             path = defaultPath;
         Properties properties = new PropReader(path).getProperties();
         if (properties.isEmpty())
             return;
         try {
             setCapabilities.execute(properties);
-        } catch (Exception ignore) { }
+        } catch (Exception ignore) {
+        }
+    }
+
+    private void setSearchStrategy(String param) {
+        String p = param.toLowerCase();
+        if ("soft".equals(p)) {
+            p = "any, multiple";
+        }
+        if ("strict".equals(p)) {
+            p = "visible, single";
+        }
+        if (p.split(",").length == 2) {
+            List<String> params = asList(p.split(","));
+            if (params.contains("visible") || params.contains("displayed")) {
+                onlyVisible();
+            }
+            if (params.contains("any") || params.contains("all")) {
+                noValidation();
+            }
+            if (params.contains("enabled")) {
+                visibleEnabled();
+            }
+            if (params.contains("inview")) {
+                inView();
+            }
+            if (params.contains("single")) {
+                STRICT_SEARCH = true;
+            }
+            if (params.contains("multiple")) {
+                STRICT_SEARCH = false;
+            }
+        }
     }
 
-    private static Properties getProperties(String path) {
+    private PageLoadStrategy getPageLoadStrategy(String strategy) {
+        switch (strategy.toLowerCase()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5482899)